addons/project_purchase/models/project_project.py

```diff
@@ -65,27 +65,24 @@ def action_open_project_purchase_orders(self):
 
     def action_profitability_items(self, section_name, domain=None, res_id=False):
         if section_name == 'purchase_order':
```

Comment:
but i still think, we can keep the old logic instead having 2 dict, to avoid missing some attributes, like domain and views when you moved it here

this is my suggestion
- we can delete many extra tabs
- when you removed the old code, you removed 'views': [False, 'form'] when res_id is set
- you can add limit = 1 when searching to improve the search as you know only one record exist
- domain also disappears
- context for me is readable better in many lines, especially if we need to add new attributes in the future.

```
if section_name != 'purchase_order':
    return super().action_profitability_items(section_name, domain, res_id)

res = {
     'name': self.env._('Purchase Order Items'),
     'res_model': 'purchase.order.line',
     'views': [[False, 'list'], [False, 'form']],
    'type': 'ir.actions.act_window',
    'domain': domain,
     'context': {
          'create': False,
          'edit': False,
      }
} 

if res_id:
    res.update({
        'name': self.env._('Purchase Orders'),
        'res_model': 'purchase.order',
        'views': [[False, 'form']],
        'view_mode': 'form',
        'res_id': self.env['purchase.order'].search([('order_line', '=', res_id)], limit = 1).id,
   } 

return res      
```

Reply:
> but i still think, we can keep the old logic instead having 2 dict, to avoid missing some attributes, like domain and views when you moved it here
> 
> this is my suggestion
> 
> * we can delete many extra tabs
> * when you removed the old code, you removed 'views': [False, 'form'] when res_id is set
> * you can add limit = 1 when searching to improve the search as you know only one record exist
> * domain also disappears
> * context for me is readable better in many lines, especially if we need to add new attributes in the future.
> 
> ```
> if section_name != 'purchase_order':
>     return super().action_profitability_items(section_name, domain, res_id)
> 
> res = {
>      'name': self.env._('Purchase Order Items'),
>      'res_model': 'purchase.order.line',
>      'views': [[False, 'list'], [False, 'form']],
>     'type': 'ir.actions.act_window',
>     'domain': domain,
>      'context': {
>           'create': False,
>           'edit': False,
>       }
> } 
> 
> if res_id:
>     res.update({
>         'name': self.env._('Purchase Orders'),
>         'res_model': 'purchase.order',
>         'views': [[False, 'form']],
>         'view_mode': 'form',
>         'res_id': self.env['purchase.order'].search([('order_line', '=', res_id)], limit = 1).id,
>    } 
> 
> return res      
> ```

1) Limit: It is necessary to pass a limit.
2) Domain Passing: If I have the ID of the purchase order line, do I still need to pass a domain?
3) Views: I have passed the view_mode, so we don't need to specify views. I am sharing the link below where it 
     is not passed. [link](https://github.com/odoo/odoo/blob/18.0/addons/account_debit_note/wizard/account_debit_note.py#L88:L9)

Reply:
1- LIMIT 1 can help the database stop scanning once it finds one result, slightly improving performance.
2- you're right, you can remove it
3- others are putting it https://github.com/odoo/enterprise/blob/acdcfc3448ab9272b04c5e8f8b78d8aba368eb4d/l10n_au_hr_payroll/wizards/l10n_au_termination_payment.py#L93-L94

https://github.com/odoo/enterprise/blob/acdcfc3448ab9272b04c5e8f8b78d8aba368eb4d/l10n_au_hr_payroll/wizards/l10n_au_termination_payment.py#L93-L94

https://github.com/odoo/enterprise/blob/3c64aa6dcd191248f0806c0b84ffa416d455fde4/appointment/static/src/views/gantt/gantt_renderer.js#L129-L130

there are many other examples

TBH, i don't really know the reason but maybe to be sure that view_id = false and be explicit about it.

Reply:
Hello @RaoufGhrissi :wave: 

Thanks for the review :sparkles: 

I've made the updates as you suggested. I've restored 'views': [[False, 'form']] when res_id is set and added limit=1.

The reason we didn't use the update method is that we are updating every value in the dictionary except for `context`.
Instead of using update, we opted to set each key explicitly.

What do you think? Which approach do you believe is the best way to handle this?
Have a nice day :innocent:

Reply:
@RaoufGhrissi Done with the changes...
And updated the commit message too.. :100: 

Thanks for the review  :innocent: